### ISSUE TITLE
Services won't start on new builds of Nethunter (e.g. 3.15.4) on OnePlus One

### ISSUE THREAD

**Issue by electricworry** [offensive-security/nethunter-app#164]
The issue I have is that services (such as SSH, Metasploit) never start automatically on device boot. If I understand correctly, these services should be started by the NetHunter app on device boot. However, when I look at the app permissions, the app has the "Auto start" permission, but it is "(never used)". Despite several reboots, I never see the services start, and I don't get the nice messages on startup like "RUNNING BOOT SERVICES OK".

Note, that this definitely used to work with the release build 3.0. I can confirm that the following combination of stock image, and NetHunter worked:
* cm-12.1-YOG7DAS2K1-bacon-signed-fastboot.zip
* nethunter-oneplus1-lollipop-3.0.zip

However, when I use the nightly files, it doesn't:
* cm-12.1-YOG7DAS2K1-bacon-signed-fastboot.zip
* nethunter-generic-armhf-kalifs-full-rolling-3.15.4-20170211-0304.zip
* kernel-nethunter-oneplus1-lollipop-3.15.4-20170211-0304.zip

I've also tried removing NetHunter app and reinstalling using my own build from this repository. Unfortunately, that doesn't help.

Any suggestions? I'm not an Android developer but I'm keen to learn. I can see that other apps (such as Email) do use the Auto Start permission. I'm assuming that this is an app issue, but am I wrong on this; i.e. could it be a kernel issue?

Thanks.

**Comment by electricworry** [comment]
I think I've figured it out...

At some point the following change was made:

```
$ git diff 5a69dd5ca35ed3e3180e72ef65a62f0b5a5556b3 8fb473a665b6d7c22db8dea1b765f69a221642e1 AndroidManifest.xml
...
-        <receiver android:name=".receiver.BootCompletedReceiver">
+
+        <updateReceiver android:name=".updateReceiver.BootCompletedReceiver">
             <intent-filter>
                 <action android:name="android.intent.action.BOOT_COMPLETED" />
             </intent-filter>
-        </receiver>
+        </updateReceiver>
...
```

Reversing this change allows Auto Start to work again.

Would it be helpful for me to fork and offer a pull request? I'm not sure whether this change was intentional or was a replace operation that went too far.

**Comment by an0mali** [comment]
It doesn't seem that code in particular at fault, but the Compile SDK version. At that point it went from API23 to API25, which doesn't seem to work properly on CM12.1 for some reason. Readme.md has not been updated to reflect this change, not sure if it was intentional? Trying to recompile it with API23, on my machine, always results in:

`/home/blah/Documents/apk_source/nh.newest.dev/nethunter-app (copy)/build/intermediates/res/merged/debug/values-v24/values-v24.xml
Error:(3) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Borderless.Colored'.
Error:(4) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Colored'.
Error:(3) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Borderless.Colored'.
Error:(4) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Colored'.`

The above appears to be caused by a commit changing the theme to reference private resources?

...But honestly... I have no idea what I'm doing. I just built a custom kernel for m7spr cm12.1 using a hybrid of original and GPE drivers to create a custom kernel to run NH... got it all up and doing neat stuff, but ended up diving into this app's source code when boot up scripts and such weren't working; which is how I stumbled upon this thread.

Origin-master is still compiled with API23, but lacks latest commits. It appears GPS, GPSD, NMEA function but the code to pass the updates on to chroot hasn't been implemented on that fork... I'm working on Frankenstein all the code meow. Not familiar with git's commit system, however...

So much fun, so little time.

**Comment by an0mali** [comment]
SO...

`android:TextAppearance.Material.Widget.Button.Borderless.Colored"`
and
'android:TextAppearance.Material.Widget.Button.Colored'

Requires minimum API level of 24, and Android 5.1.1 is API23, so backwards compat is broken at this point in time for master fork... I think...

**Comment by Re4son** [comment]
Thanks for raising this issue.
Should be fixed in v2019.1.
